From 7862379f3fb44780b1113da7a67eeebe5238348b Mon Sep 17 00:00:00 2001
From: Jon Hefling <jon.hefling@lairdtech.com>
Date: Thu, 12 Nov 2015 09:09:26 -0600
Subject: [PATCH] Laird: wb50n: Enable support for NAT/Masquerade.

Enable iproute2/netfilter options for iptables Masquerade.
Fixed related nf_conntrack options to match wb45n.

Bug 8568.
---
 arch/arm/configs/wb50n_defconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/configs/wb50n_defconfig b/arch/arm/configs/wb50n_defconfig
index 254e8708..3d42f1f 100644
--- a/arch/arm/configs/wb50n_defconfig
+++ b/arch/arm/configs/wb50n_defconfig
@@ -58,6 +58,12 @@ CONFIG_INET_UDP_DIAG=m
 CONFIG_TCP_MD5SIG=y
 CONFIG_NETFILTER=y
 CONFIG_NETFILTER_DEBUG=y
+CONFIG_NF_CONNTRACK=m
+# CONFIG_NF_CONNTRACK_PROCFS is not set
+CONFIG_NF_CONNTRACK_EVENTS=y
+CONFIG_NF_CONNTRACK_FTP=m
+CONFIG_NF_CONNTRACK_TFTP=m
+CONFIG_NF_CT_NETLINK=m
 CONFIG_NF_TABLES=m
 CONFIG_NF_TABLES_INET=m
 CONFIG_NFT_EXTHDR=m
@@ -68,8 +74,11 @@ CONFIG_NFT_COUNTER=m
 CONFIG_NFT_LOG=m
 CONFIG_NFT_LIMIT=m
 CONFIG_NFT_REJECT=m
+CONFIG_NF_CONNTRACK_IPV4=m
 CONFIG_IP_NF_IPTABLES=y
 CONFIG_IP_NF_FILTER=y
+CONFIG_IP_NF_NAT=m
+CONFIG_IP_NF_TARGET_MASQUERADE=m
 CONFIG_IP_NF_ARPTABLES=y
 CONFIG_IP_NF_ARPFILTER=y
 CONFIG_IP_NF_ARP_MANGLE=y
-- 
1.9.1

